cirq-core/cirq/ops/raw_types.py

```diff
@@ -630,16 +630,19 @@ def with_classical_controls(
         measurement key. A measurement key evaluates to True iff any qubit in
         the corresponding measurement operation evaluated to a non-zero value.
 
-        The classical control will hide any tags on the existing operation,
-        since tags are considered a local attribute.
+        If no conditions are specified, returns self.
+
+        The classical control will remove any tags on the existing operation,
+        since tagged operations are considered to be immutable.
```

Comment:
The reason isn't that TaggedOperations are immutable, as this creates a new operation - no mutation is taking place regardless. The actual reason is that tags are fragile - for an arbitrary tag, it's not possible to determine whether it should be preserved after a change, so we opt to always discard them.

Reply:
Yeah, that's why I tried writing "considered" to be immutable. 

How about I reword it to?

```
The classical control will remove any tags on the existing operation,
since tags are fragile, and we always opt to get rid of the tags when
the underlying operation is changed.
```

Any suggestions with the right wording are welcome.

Reply:
That looks good to me. Granting approval with the assumption that this gets applied.